### PR TITLE
docs: migration doc updates for 08-wasm v8 to v9

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

closes: #6139 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 1]
```diff
@@ -210,9 +210,20 @@ The `IterateConsensusMetadata` function has been removed.
 
 ### 08-wasm
 
+- The `Initialize`, `Status`, `GetTimestampAtHeight`, `GetLatestHeight`, `VerifyMembership`, `VerifyNonMembership`, `VerifyClientMessage`, `UpdateState` and `UpdateStateOnMisbehaviour` functions in `ClientState` have been removed and all their logic has been moved to functions of the `LightClientModule`.
+- The `MigrateContract()` function has been removed from `ClientState`.
```
Comment (crodriguezvega) [inline docs/docs/05-migrations/13-v8-to-v9.md:214]: nit: I don't feel strongly about using `()` when we refer to functions, but for consistency with the rest of the document we should either remove them or add them to all.

**Review by crodriguezvega — COMMENTED** [review]
This looks good; thank you, @gjermundgaraba! We should move the whole `08-wasm` section to the specific [migration docs file for `08-wasm`](https://github.com/cosmos/ibc-go/blob/main/docs/docs/03-light-clients/04-wasm/09-migrations.md). We can have a heading like `From ibc-go v8.3.x to ibc-go v9.0.x`

File: docs/docs/05-migrations/13-v8-to-v9.md (hunk 1 above)
Comment (damiannolan) [reply docs/docs/05-migrations/13-v8-to-v9.md:214]: Fine with removing the parentheses  👍🏻

**Review by damiannolan — APPROVED** [review]

File: docs/docs/03-light-clients/04-wasm/09-migrations.md [hunk 2]
```diff
@@ -9,6 +9,25 @@ slug: /ibc/light-clients/wasm/migrations
...
+
+### Chains
+
+- The `Initialize`, `Status`, `GetTimestampAtHeight`, `GetLatestHeight`, `VerifyMembership`, `VerifyNonMembership`, `VerifyClientMessage`, `UpdateState` and `UpdateStateOnMisbehaviour` functions in `ClientState` have been removed and all their logic has been moved to functions of the `LightClientModule`.
+- The `MigrateContract` function has been removed from `ClientState`.
+- The `VerifyMembershipMsg` and `VerifyNonMembershipMsg` payloads for `SudoMsg` have been modified. The `Path` field of both structs has been updated from `v1.MerklePath` to `v2.MerklePath`. The new `v2.MerklePath` field contains a `KeyPath` of `[][]byte` as opposed to `[]string`, see [23-commitment](#23-commitment). This supports proving values stored under keys which contain non-utf8 encoded symbols. As a result, the JSON field `path` containing `key_path` of both messages will marshal elements as a base64 encoded bytestrings. This is a breaking change for 08-wasm client contracts and they should be migrated to correctly support deserialisation of the `v2.MerklePath` field. See [08-wasm migrations](../03-light-clients/04-wasm/09-migrations.md) for more information.
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply docs/docs/03-light-clients/04-wasm/09-migrations.md:18]: Oh, that was what it targeted. Yeah that makes sense. Will fix that 🙈

**Review by crodriguezvega — APPROVED** [review]
We can merge this one, right?

**Pull request merged**